api/src/main/java/openconsensus/metrics/MetricCollection.java

```diff
@@ -104,4 +103,37 @@
    */
   public abstract DerivedDoubleCumulative addDerivedDoubleCumulative(
       String name, MetricOptions options);
+
+  /** Builder class for the {@link MetricCollection}. */
+  public abstract static class Builder {
+
+    /**
+     * Sets the name of the component that reports these metrics.
+     *
+     * <p>The final name of the reported metric will be <code>component + "_" + name</code> if the
+     * component is not empty.
+     *
+     * @param component the name of the component that reports these metrics.
+     * @return this.
+     */
+    public abstract Builder setComponent(String component);
```

Comment:
hm... is there any relationship of `component` and `Resource`?

Reply:
perhaps mention it in summary as minimum

Reply:
* Component: will be something like "apache_kafka" - it is important when do instrumentation.
* Resource: it is used when reporting out-of-band metrics, for example a sidecar reading logs and generating metrics on behalf of the task that generated the logs.